src/main/webapp/2Dscript.js

```diff
@@ -80,7 +180,410 @@ function addCenterControl(controlDiv) {
+    const subSectionNumber = 1;
+    // Clear prior paths, only display newly selected ones.
+    if (markerBottom !== undefined) {
+      markerBottom.setMap(null);
+    }
+    if (markerTop !== undefined) {
+      markerTop.setMap(null);
+    }
+    if (subPath !== undefined) {
+      subPath.setMap(null);
+    }
+    subPath = getPolyLine(subLine, 'blue', 1.0, 2);
+    // Setup event listener to show option for 3D window when polyline is
+    // clicked.
+    google.maps.event.addListener(subPath, 'click', function(event) {
+      const latLng = event.latLng;
+      console.log('Polyline clicked at lat: ' + latLng.lat() +
+        ' lng: ' + latLng.lng());
+      infoWindow = new google.maps.InfoWindow({
+        content: '<a href=/home.html?id=' + id + '&dataType=' +
+          type + '&subSection=' + subSectionNumber +
+          '&stored=true' + '> View in 3D </a>',
+        position: latLng,
+      });
+      infoWindow.setMap(map);
+    },
+    );
+
+    subPath.setMap(map);
+
+    markerBottom = genMarker(discoveredMinLat, discoveredMinLatPair);
+    markerTop = genMarker(discoveredMaxLngPair, discoveredMaxLng);
+    markerBottom.setMap(map);
+    markerTop.setMap(map);
+
+    sessionStorage.setItem(id + '_' + type + '_' + subSectionNumber,
+        JSON.stringify(subLine));
+  }
+}
+
+/**
+ * Change the rectangle width to be match the current cursor position.
+ * @param {MouseEvent} event
+ */
+function genChangingBox(event) {
+  if (isMouseDown) {
+    console.log('Set width triggered');
+    finX = event.pageX;
+    finY = event.pageY;
+    widget.width(finX - x);
+    widget.height(finY - y);
+    widget.css({
+      'width': (finX - x) + '!important',
+      'height': (finY - y) + '!important',
+      'display': 'block',
+      'border': '2px dashed #ccc',
+    });
+  }
+}
+
+/**
+ * Generate a Google maps marker at the given lat and long.
+ * @param {number} latitude
+ * @param {number} longitude
+ * @return {google.maps.Marker}
+ */
+function genMarker(latitude, longitude) {
+  const marker = new google.maps.Marker({
+    position: {lat: latitude, lng: longitude},
+    title: 'Lat: ' + latitude + ' Lng: ' + longitude,
+  });
+  return marker;
+}
+
+
+/**
+ * Create a polyline from the given points with specified parameters.
+ * @param {Point} linePoints
+ * @param {string} color
+ * @param {number} opacity
+ * @param {number} weight
+ * @param {number} index
+ * @return {google.maps.Polyline}
+ */
+function getPolyLine(linePoints, color, opacity, weight, index = 1) {
+  const polyLine = new google.maps.Polyline({
+    path: linePoints,
+    geodesic: true,
+    strokeColor: color,
+    strokeOpacity: opacity,
+    strokeWeight: weight,
+    zIndex: index,
+  });
+  return polyLine;
+}
+
+/**
+ * Compute the subsection of a line contained in a bounding rectangle.
+ * @param {Point} pose
+ * @param {number} currentLat
+ * @param {number} priorLat
+ * @param {number} currentLng
+ * @param {number} priorLng
```

Comment:
I think `pose` is actually an Array? And current/prior is surprising terminology for a bounding rectangle; I would expect something like min/max.

Reply:
Yeah should be Array<Point> good catch. The reasoning for this was the first click gives the priorLat and priorLng and the next click gives the currentLng and currentLng that you clicked to. That's why I used that terminology as I was thinking in terms of when the events happen not the bounds of the rectangle itself.